hsmd/hsmd.c

```diff
@@ -61,29 +62,20 @@
  * stream from lightningd. */
 #define REQ_FD 3
 
-/*~ Nobody will ever find it here!  hsm_secret is our root secret, the bip32
- * tree and bolt12 payer_id keys are derived from that, and cached here. */
-static struct {
-	struct secret hsm_secret;
-	struct ext_key bip32;
-	secp256k1_keypair bolt12;
-} secretstuff;
-
-/* Have we initialized the secretstuff? */
-static bool initialized = false;
-
 /* Version codes for BIP32 extended keys in libwally-core.
  * It's not suitable to add this struct into client struct,
  * so set it static.*/
-static struct  bip32_key_version  bip32_key_version;
+extern struct  bip32_key_version  bip32_key_version;
```

Comment:
extern in a C file is an anti-pattern; we usually want it in a header so both the user and implementer are *definitely* using the same type...

Reply:
Yeah, I was hoping I could just move everything across and then remove the `extern` from `hsmd.c` but since we still do some operations in `hsmd.c` like overriding the privkey we need to keep a way to reference it. I could move it to the `libhsmd.h` file, but that breaks the encapsulation. Maybe we find a good way to move it into libhsmd.